onnxruntime/core/providers/cpu/ml/svmclassifier.h

```diff
@@ -30,31 +30,31 @@ class SVMCommon {
   KERNEL get_kernel_type() const { return kernel_type_; }
 
   float kernel_dot(const T* A, int64_t a, const std::vector<float>& B, int64_t b, int64_t len, KERNEL k) const {
-    float sum = 0.f;
+    double sum = 0.f;
+    const T* pA = A + a;
+    const float* pB = &B[b];
```

Comment:
>&B[b] [](start = 22, length = 5)

nit: Maybe `B.data()+b` to avoid the creation of the temporal object `B[a]`?

Reply:
done